### PR TITLE
Update Axes showcase in "Embedding in Tk" example

### PR DESCRIPTION
**Pull request by kopytjuk**
## PR Summary

From the example in https://matplotlib.org/stable/gallery/user_interfaces/embedding_in_tk_sgskip.html it is not clear how to update plots in interactive user interfaces (see https://github.com/matplotlib/matplotlib/issues/20159). This PR extends the example with an *update* button which shows how to update the plot in the `FigureCanvasTkAgg`.

image

## PR Checklist

- [ N/A] Has pytest style unit tests (and `pytest` passes).
- [ N/A] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [N/A] New features are documented, with examples if plot related.
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [x] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [N/A] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [N/A] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: examples/user_interfaces/embedding_in_tk_sgskip.py [hunk 1]
```diff
@@ -30,18 +31,35 @@
...
+    # retrieve Axes object and update plots there
+    ax = fig.axes[0]
+    ax.clear()
+    ax.plot(t, 2 * np.sin(2 * np.pi * f * t))
+
+    # required to update canvas and attached toolbar!
+    canvas.draw()
+
+
+button_update = tkinter.Button(
```
Comment (jklymak) [inline examples/user_interfaces/embedding_in_tk_sgskip.py:54]: Thanks, this seems useful!  Any chance you could make this a slider instead?  If not, can you change the button name to "Change Freq" or something more indicative of what it does?

Comment (kopytjuk) [reply examples/user_interfaces/embedding_in_tk_sgskip.py]: Done, looks like this now:

image

File: examples/user_interfaces/embedding_in_tk_sgskip.py [hunk 2]
```diff
@@ -30,18 +32,39 @@
...
+    f = float(new_val)
+
+    # retrieve Axes object and update plots there
+    ax = fig.axes[0]
+    # alternatively you could delete last plotted line `del ax.lines[0]`
+    ax.clear()
+
+    ax.plot(t, 2 * np.sin(2 * np.pi * f * t))
+    #  since Axes was cleared
+    ax.set_xlabel("x")
```
Comment (QuLogic) [inline examples/user_interfaces/embedding_in_tk_sgskip.py:53]: There's likely no need to clear and re-plot everything. Save the result of `ax.plot` and call `.set_data` on it. Or if that's not something you want to show, then call `.remove` on it.

File: examples/user_interfaces/embedding_in_tk_sgskip.py (hunk 1 above)
Comment (jklymak) [reply examples/user_interfaces/embedding_in_tk_sgskip.py]: Cool.  I think if you added @QuLogic suggestion below that would be very nice as well.  What you are doing is the simplest thing to do, but what he is suggesting is "best-practice" when dealing with animations.

File: examples/user_interfaces/embedding_in_tk_sgskip.py (hunk 2 above)
Comment (kopytjuk) [reply examples/user_interfaces/embedding_in_tk_sgskip.py:53]: I will implement that for this example.

File: examples/user_interfaces/embedding_in_tk_sgskip.py [hunk 3]
```diff
@@ -21,7 +21,10 @@
 
 fig = Figure(figsize=(5, 4), dpi=100)
 t = np.arange(0, 3, .01)
-fig.add_subplot().plot(t, 2 * np.sin(2 * np.pi * t))
+ax = fig.add_subplot()
+line = ax.plot(t, 2 * np.sin(2 * np.pi * t))
```
Comment (QuLogic) [inline examples/user_interfaces/embedding_in_tk_sgskip.py:25]: We usually do:
```suggestion
line, = ax.plot(t, 2 * np.sin(2 * np.pi * t))
```
to unpack it directly. And then it'll match the name.

File: examples/user_interfaces/embedding_in_tk_sgskip.py [hunk 4]
```diff
@@ -30,18 +33,34 @@
...
+button_quit = tkinter.Button(master=root, text="Quit", command=root.quit)
+
+
+def update_frequency(new_val):
+    # retrieve frequency
+    f = float(new_val)
+
+    # update data
+    y = 2 * np.sin(2 * np.pi * f * t)
+    line[0].set_data(t, y)
```
Comment (QuLogic) [inline examples/user_interfaces/embedding_in_tk_sgskip.py:49]:
```suggestion
    line.set_data(t, y)
```

**Review by QuLogic — APPROVED** [review]

**Pull request merged**